hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -570,6 +584,9 @@ private HoodieIndexDefinition getFunctionalIndexDefinition(String indexName) {
   }
 
   private Set<String> getIndexPartitionsToInit(MetadataPartitionType partitionType) {
+    if (dataMetaClient.getIndexMetadata().isEmpty()) {
```

Comment:
```suggestion
    if (dataMetaClient.getFunctionalAndSecondaryIndexMetadata().isEmpty()) {
```

Reply:
the class is named as HoodieIndexMetadata and hence the getter. Not sure if we want to keep it generic. Will defer it in this patch. we can discuss w/ @codope and fix the naming in a follow up patch if need be.

Reply:
https://issues.apache.org/jira/browse/HUDI-8398

Reply:
The name is confusing while I was reading the code: `dataMetaClient.getIndexMetadata()` seems to indicate that this is for all index metadata, but it is only for functional and secondary index.